### PR TITLE
Follow up of #23461

### PR DESCRIPTION
**Pull request by vipulnsward**
- Rename max to statement_limit
- Remove magic number 1000 from everywhere
- Defined StatementPool::DEFAULT_STATEMENT_LIMIT and started using it everywhere

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ class StatementPool < ConnectionAdapters::StatementPool
       def initialize(connection, logger, connection_options, config)
         super(connection, logger, config)
 
-        @statements = StatementPool.new(self.class.type_cast_config_to_integer(config.fetch(:statement_limit) { 1000 }))
+        @statements = StatementPool.new(self.class.type_cast_config_to_integer(config.fetch(:statement_limit) { StatementPool::DEFAULT_STATEMENT_LIMIT }))
```
Comment (jeremy) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: We can remove the defaults from all these. `self.class.type_cast_config_to_integer(nil)` will return `nil`, so we can handle defaults in `StatementPool#initialize`.

```ruby
@statements = StatementPool.new(self.class.type_cast_config_to_integer(config[:statement_limit])
```

File: activerecord/lib/active_record/connection_adapters/statement_pool.rb [hunk 2]
```diff
@@ -3,9 +3,11 @@ module ConnectionAdapters
     class StatementPool
       include Enumerable
 
-      def initialize(max = 1000)
+      DEFAULT_STATEMENT_LIMIT = 1000
+
+      def initialize(statement_limit = DEFAULT_STATEMENT_LIMIT)
         @cache = Hash.new { |h,pid| h[pid] = {} }
-        @max = max
+        @statement_limit = statement_limit
```
Comment (jeremy) [inline activerecord/lib/active_record/connection_adapters/statement_pool.rb]:
```ruby
def initialize(statement_limit = nil)
  …
  @statement_limit ||= DEFAULT_STATEMENT_LIMIT
end
```

Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/statement_pool.rb]: Hmm, I had tried previous version, hence was not passing nil here and instead DEFAULT_STATEMENT_LIMIT. 
Let me try it out.

Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/statement_pool.rb]: Thanks, that worked.

**Comment by jeremy** [comment]
6c6eeda54d0f2c8e689772b50cc5c4419fa82749 😊

**Pull request closed without merging**